### PR TITLE
chore(policy-engine): rough docs for policy engine

### PR DESCRIPTION
**Pull request by ethanfrogers**
adds some rough documentation for the policy engine plugin. includes
install instructions as well as usage instructions.

### CODE REVIEW

File: content/en/docs/plugin-guide/plugin-policy-engine.md [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+
+TODO - figure out if there are any limitations
+-->
+
+## Setup
+
+The plugin can be delivered using two different methods:
+
+1. Docker image as an init container on each affected service
+1. Using a remote plugin repository
```
Comment (aelath) [inline content/en/docs/plugin-guide/plugin-policy-engine.md]: Shouldn't this be a `2`?
```suggestion
2. Using a remote plugin repository
```

**Review by aelath — APPROVED** [review]
LGTM! 🌮 
Just a numbering comment.

File: content/en/docs/plugin-guide/plugin-policy-engine.md (hunk 1 above)
Comment (317brian) [reply content/en/docs/plugin-guide/plugin-policy-engine.md]: The site generator auto increments numbered lists, so you could just do 1s all the way down. But yeah, it looks weird to humans when writing it lol.

**Review by 317brian — APPROVED** [review]
:blob_hype: lgtm

File: content/en/docs/plugin-guide/plugin-policy-engine.md (hunk 1 above)
Comment (ethanfrogers) [reply content/en/docs/plugin-guide/plugin-policy-engine.md]: Actually, Markdown will still render it properly. One reason to just use `1.` is so you don't have to re-number the whole list when adding a new item. :D

File: content/en/docs/plugin-guide/plugin-policy-engine.md [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+
+   ```yaml
+   spinnaker:
+     extensibility:
+       plugins:
+           Armory.PolicyEngine:
+               enabled: true
+   ```
+
+3. Add the following to `service-settings/gate.yml`, `service-settings/orca.yml`, `service-settings/clouddriver.yml` and `service-settings/clouddriver.yml`:
```
Comment (KathrynLewis) [inline content/en/docs/plugin-guide/plugin-policy-engine.md]:
```suggestion
3. Add the following to `service-settings/gate.yml`, `service-settings/orca.yml`, `service-settings/clouddriver.yml` and `service-settings/front50.yml`:
```

File: content/en/docs/plugin-guide/plugin-policy-engine.md [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+
+4. Add the following to  `.hal/config`:
+
+   ```yaml
+   deploymentConfigurations:
+     - name: default
+       deploymentEnvironment:
+         initContainers:
+           front50:
+            - name: policy-engine-install
```
Comment (KathrynLewis) [inline content/en/docs/plugin-guide/plugin-policy-engine.md]:
```suggestion
             - name: policy-engine-install
```

File: content/en/docs/plugin-guide/plugin-policy-engine.md [hunk 4]
```diff
@@ -0,0 +1,450 @@
...
+    stage := canDeployStages[_]
+    not stage.context.canDeploy == true
+}
+```
+
+### Pipeline save
+
+This is the simplest use case for Policy Engine and where many users begin. You can enforce policies on all pipelines when they get saved. This ensures that any pipeline that violates a policy cannot be saved. A common use case for this kind of policy is requiring all pipelines to have a certain stage. T
+
+he following example shows the `rego` syntax for a policy that requires all pipelines to have a `preFlightCheck` type stage:
```
Comment (KathrynLewis) [inline content/en/docs/plugin-guide/plugin-policy-engine.md]:
```suggestion
This is the simplest use case for Policy Engine and where many users begin. You can enforce policies on all pipelines when they get saved. This ensures that any pipeline that violates a policy cannot be saved. A common use case for this kind of policy is requiring all pipelines to have a certain stage. 

The following example shows the `rego` syntax for a policy that requires all pipelines to have a `preFlightCheck` type stage:
```

File: content/en/docs/plugin-guide/plugin-policy-engine.md [hunk 5]
```diff
@@ -0,0 +1,450 @@
...
+
+   ```yaml
+   deploymentConfigurations:
+     - name: default
+       deploymentEnvironment:
+         initContainers:
+           front50:
+            - name: policy-engine-install
+               image: docker.io/armory/policy-engine-plugin:<PLUGIN VERSION>
+               volumeMounts:
```
Comment (KathrynLewis) [inline content/en/docs/plugin-guide/plugin-policy-engine.md]:
```suggestion
               image: docker.io/armory/policy-engine-plugin:<PLUGIN VERSION>
               args:
                 - -install-path
                 - /opt/policy-engine-plugin/target
               volumeMounts:
```

File: content/en/docs/plugin-guide/plugin-policy-engine.md [hunk 6]
```diff
@@ -0,0 +1,450 @@
...
+                 - mountPath: /opt/policy-engine/target
+                   name: policy-engine-install
+           clouddriver:
+             - name: policy-engine-install
+               image: docker.io/armory/policy-engine-plugin:<PLUGIN VERSION>
+               volumeMounts:
```
Comment (KathrynLewis) [inline content/en/docs/plugin-guide/plugin-policy-engine.md]:
```suggestion
               image: docker.io/armory/policy-engine-plugin:<PLUGIN VERSION>
               args:
                 - -install-path
                 - /opt/policy-engine-plugin/target
               volumeMounts:
```

File: content/en/docs/plugin-guide/plugin-policy-engine.md [hunk 7]
```diff
@@ -0,0 +1,450 @@
...
+                 - mountPath: /opt/policy-engine/target
+                   name: policy-engine-install
+           gate:
+             - name: policy-engine-install
+               image: docker.io/armory/policy-engine-plugin:<PLUGIN VERSION>
+               volumeMounts:
```
Comment (KathrynLewis) [inline content/en/docs/plugin-guide/plugin-policy-engine.md]:
```suggestion
               image: docker.io/armory/policy-engine-plugin:<PLUGIN VERSION>
               args:
                 - -install-path
                 - /opt/policy-engine-plugin/target
               volumeMounts:
```

File: content/en/docs/plugin-guide/plugin-policy-engine.md [hunk 8]
```diff
@@ -0,0 +1,450 @@
...
+                 - mountPath: /opt/policy-engine/target
+                   name: policy-engine-install
+           orca:
+             - name: policy-engine-install
+               image: docker.io/armory/policy-engine-plugin:<PLUGIN VERSION>
+               volumeMounts:
```
Comment (KathrynLewis) [inline content/en/docs/plugin-guide/plugin-policy-engine.md]:
```suggestion
               image: docker.io/armory/policy-engine-plugin:<PLUGIN VERSION>
               args:
                 - -install-path
                 - /opt/policy-engine-plugin/target
               volumeMounts:
```

**Review by KathrynLewis — APPROVED** [review]

**Pull request merged**